bom/build.gradle.kts

```diff
@@ -21,3 +45,109 @@ dependencies {
         api(project(":driver-scala"))
     }
 }
+
+/*
+ * Handle the multiple versions of Scala we support as defined in `gradle.properties`
+ */
+val defaultScalaVersion: String = project.findProperty("defaultScalaVersion")!!.toString()
+val scalaVersions: List<String>? = project.findProperty("supportedScalaVersions")?.toString()?.split(",")
+
+assert(!scalaVersions.isNullOrEmpty()) {
+    "Scala versions must be provided as a comma-separated list in the 'supportedScalaVersions' project property"
+}
+
+/*
+ * Apply the Java Platform plugin to create the BOM
+ * Modify the generated POM to include all supported versions of Scala for driver-scala or bson-scala.
+ */
+configureMavenPublication {
+    components.findByName("javaPlatform")?.let { from(it) }
+
+    pom {
+        name.set("bom")
+        description.set(
+            "This Bill of Materials POM simplifies dependency management when referencing multiple MongoDB Java Driver artifacts in projects using Gradle or Maven.")
+
+        withXml {
+            val pomXml: Node = asNode()
+
+            val dependencyManagementNode = pomXml.getNode("dependencyManagement")
+            assert(dependencyManagementNode != null) {
+                "<dependencyManagement> node not found in the generated BOM POM"
+            }
+            val dependenciesNode = dependencyManagementNode.getNode("dependencies")
+            assert(dependenciesNode != null) { "<dependencies> node not found in the generated BOM POM" }
+
+            val existingScalaDeps =
+                dependenciesNode!!
+                    .children()
+                    .map { it as Node }
+                    .filter { it.getNode("artifactId")?.text()?.contains("scala") ?: false }
+
+            existingScalaDeps.forEach {
+                val groupId: String = it.getNode("groupId")!!.text()
+                val originalArtifactId: String = it.getNode("artifactId")!!.text()
+                val artifactVersion: String = it.getNode("version")!!.text()
+
+                // Add multiple versions with Scala suffixes for each Scala-related dependency.
+                scalaVersions!!.forEach { scalaVersion ->
+                    if (scalaVersion != defaultScalaVersion) {
+                        // Replace scala version suffix
```

Comment:
super minor. `scala version` -> `Scala version`? Don't wanna make a fuss about it. Feel free to ignore.